### PR TITLE
SQLite integration

### LINKED ISSUE #463: Integrate FerretDB testing with SQLite
Let's start by thinking (or completely rethinking) the YAML structure. Could we use YAML anchors or something? or just some common section?

### PR DESCRIPTION
**Pull request by b1ron**
Closes #463.

### CODE REVIEW

File: internal/configload/configload.go [hunk 1]
```diff
@@ -36,20 +36,23 @@ type config struct {
...
-	Skip    []any      `yaml:"skip"`
-	Pass    []any      `yaml:"pass"`
-	Ignore  []any      `yaml:"ignore"`
+	Default     *ic.Status `yaml:"default"`
+	Stats       *stats     `yaml:"stats"`
+	Fail        []any      `yaml:"fail"`
+	Skip        []any      `yaml:"skip"`
+	Pass        []any      `yaml:"pass"`
+	Ignore      []any      `yaml:"ignore"`
+	IncludeFail []any      `yaml:"include_fail"`
```
Comment (AlekSi) [inline internal/configload/configload.go:55]: I wonder if all those can and should be `[]string`

File: internal/config/config.go [hunk 2]
```diff
@@ -70,6 +70,7 @@ type Config struct {
 // Results stores the expected test results for different databases.
 type Results struct {
 	FerretDB *TestConfig
```
Comment (AlekSi) [inline internal/config/config.go:72]: Let's rename that to `PostgreSQL`

File: internal/configload/configload.go [hunk 3]
```diff
@@ -36,20 +36,23 @@ type config struct {
 	Dir     string        `yaml:"dir"`
 	Args    []string      `yaml:"args"`
 	Results struct {
-		Common   *testConfig `yaml:"common"`
-		FerretDB *testConfig `yaml:"ferretdb"`
-		MongoDB  *testConfig `yaml:"mongodb"`
+		Includes map[string][]string `yaml:"includes"`
+		Common   *testConfig         `yaml:"common"`
+		FerretDB *testConfig         `yaml:"ferretdb"`
```
Comment (AlekSi) [inline internal/configload/configload.go:41]:
```suggestion
		PostgreSQL *testConfig         `yaml:"ferretdb"`
```

Preserve the YAML tag for compatibility for now. We will update it later everywhere.

File: internal/configload/configload.go [hunk 4]
```diff
@@ -36,20 +36,23 @@ type config struct {
 	Dir     string        `yaml:"dir"`
 	Args    []string      `yaml:"args"`
 	Results struct {
-		Common   *testConfig `yaml:"common"`
-		FerretDB *testConfig `yaml:"ferretdb"`
-		MongoDB  *testConfig `yaml:"mongodb"`
+		Includes map[string][]string `yaml:"includes"`
+		Common   *testConfig         `yaml:"common"`
```
Comment (AlekSi) [inline internal/configload/configload.go:40]: Let's create a separate issue for removing that and add `// TODO <url>` comment

File: internal/configload/configload.go (hunk 1 above)
Comment (b1ron) [reply internal/configload/configload.go:55]: We cannot do that unfortunately as we use nested mappings for regex patterns.

Comment (b1ron) [reply internal/configload/configload.go:55]: Unless I just use the same approach and unmarshal them into a field called `IncludeRegexp`. I will see.

Comment (b1ron) [reply internal/configload/configload.go]: I added an `IncludeSkip` field to simplify how we handle regex patterns. But [diff.yml](https://github.com/FerretDB/dance/blob/main/tests/diff.yml) now breaks as we do not use type assertion so we cannot have nested mappings.

I propose we attempt to change the diff config in the near future instead. @AlekSi what do you think?

**Review by AlekSi — CHANGES_REQUESTED** [review]
Runs fail hard: https://github.com/FerretDB/dance/actions/runs/6271907108/job/17032412184?pr=584
They should not

**Comment by b1ron** [comment]
> Runs fail hard: https://github.com/FerretDB/dance/actions/runs/6271907108/job/17032412184?pr=584
They should not

@AlekSi Fixed.

**Review by AlekSi — APPROVED** [review]

**Pull request merged**